File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/state/orderstate/OrderCheckState.java
// OrderCheckState.java

package com.bigboxcode.designpattern.state.orderstate;

public class OrderCheckState extends OrderState {
    public OrderCheckState(OrderContext context) {
        super(context);
    }

    @Override
    public void process() {
        // Write code to process the order
        System.out.println("Checking the order validity and other information");

        context.setState(context.getOrderInProgressState());
    }
}

File: design-patterns/php/state/order/OrderCheckState.php
<?php
// OrderCheckState.php

namespace BigBoxCode\DesignPattern\State\Order;


class OrderCheckState extends OrderState {
    public function __construct(OrderContext $context) {
        parent::__construct($context);
    }

    public function process(): void {
        // Write code to process the order
        echo "Checking the order validity and other information\n";

        $this->context->setState($this->context->getOrderInProgressState());
    }
}